AWS: Move enforcement of attached AWS device limit from kubelet to scheduler

jsafrane opened the pull request:
Limit of nr. of attached EBS volumes to a node is now enforced by scheduler. It can be adjusted by `KUBE_MAX_PD_VOLS` env. variable there. Therefore we don't need the same check in kubelet. If the system admin wants to attach more, we should allow it.

Kubelet limit is now 650 attached volumes ('ba'..'zz').

Note that the scheduler counts only _pods_ assigned to a node. When a pod is deleted and a new pod is scheduled on a node, kubelet start (slowly) detaching the old volume and (slowly) attaching the new volume. Depending on AWS speed **it may happen that more than KUBE_MAX_PD_VOLS volumes are actually attached to a node for some time!** Kubelet will clean it up in few seconds / minutes (both attach/detach is quite slow).

Fixes #22994

jsafrane commented:
Fixes #22994

jsafrane commented:
@justinsb @kubernetes/rh-storage, please review.

eparis commented:
@jsafrane edit the original comment to include the `fixes #23254` line.

eparis commented on pkg/cloudprovider/providers/aws/aws.go:
```diff
@@ -1050,19 +1050,20 @@ func (self *awsInstance) getMountDevice(volumeID string, assign bool) (assigned
 	}
 
-	// Check all the valid mountpoints to see if any of them are free
-	valid := instanceType.getEBSMountDevices()
```
Is there another caller of `getEBSMountDevices()`  or should you be deleting that code as well?

justinsb replied:
Was going to say the same thing - it makes sense to inline it because the volume list is now pretty big, but do delete getEBSMountDevices

jsafrane replied:
good point, removed

jsafrane commented:
TravisCI error:
```
Unable to find image 'gcr.io/google_containers/gen-swagger-docs:v5' locally
v5: Pulling from google_containers/gen-swagger-docs
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

Looks like flake.

jsafrane committed:
Remove limit of attached AWS devices from kubelet.

Limit of nr. of attached EBS volumes to a node is now enforced by scheduler. It
can be adjusted by KUBE_MAX_PD_VOLS env. variable there.

Therefore we don't need the same check in kubelet. If the system admin wants to
attach more, we should allow it.

Kubelet limit is now 650 attached volumes ('ba'..'zz').

jsafrane commented:
@justinsb, can you please take a look?

pmorie commented:
cc @DirectXMan12

justinsb commented:
Sorry for the delay @jsafrane .  I changed the title to be a clearer (?) release note - feel free to tweak!

justinsb commented:
@k8s-bot unit test this issue: #25539

ArtfulCoder commented:
@k8s-bot test this issue: #IGNORE

jeremyeder commented:
@jsafrane thank you!!! /cc @ekuric
